sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -42,7 +53,7 @@ def match_serialized_pipelineparam(payload: str):
   for match in matches:
       pattern = '{{pipelineparam:op=%s;name=%s}}' % (match[0], match[1])
       param_tuples.append(PipelineParamTuple(
-                          name=sanitize_k8s_name(match[1]), 
+                          name=sanitize_k8s_name(match[1], True),
                           op=sanitize_k8s_name(match[0]), 
```

Comment:
I'm not sure this is a proper place for sanitization. The names should probably be converted in the compiler: https://github.com/kubeflow/pipelines/blob/735e627a03629ba72a11ca89f608c1979e0c02c0/sdk/python/kfp/compiler/_op_to_template.py#L43

Reply:
Agreed. I was curious why there are two sanitize functions as well.